.github/docker-performance-tests/alarms-poller/poll-during-performance-tests.py

```diff
@@ -0,0 +1,288 @@
+                    "Dimensions": [
+                        {
+                            "Name": PROCESS_COMMAND_LINE_DIMENSION_NAME,
+                            "Value": args.process_command_line_dimension_value,
+                        }
+                    ],
+                },
+                "Stat": METRIC_DATA_STATISTIC,
+                "Period": args.metrics_period,
+            },
+            "Label": "CPU Time Raw",
+            "ReturnData": False,
+        },
+        {
+            "Id": "cpu_load_expr",
+            "Expression": f"cpu_time_raw/PERIOD(cpu_time_raw)/{args.num_of_cpus}*100",
+            "Label": f"CPU Load Percentage for {args.num_of_cpus} CPUs",
+            "ReturnData": True,
+            "Period": args.metrics_period,
+        },
+    ]
+
+    total_memory_metric_data_queries = [
+        {
+            "Id": "virtual_memory_raw",
+            "MetricStat": {
+                "Metric": {
+                    "Namespace": args.logs_namespace,
+                    "MetricName": "process.memory.virtual_usage",
+                    "Dimensions": [
+                        {
+                            "Name": PROCESS_COMMAND_LINE_DIMENSION_NAME,
+                            "Value": args.process_command_line_dimension_value,
+                        }
+                    ],
+                },
+                "Stat": METRIC_DATA_STATISTIC,
+                "Period": args.metrics_period,
+            },
+            "Label": "Virtual Memory",
+            "ReturnData": False,
+        },
+        {
+            "Id": "physical_memory_raw",
+            "MetricStat": {
+                "Metric": {
+                    "Namespace": args.logs_namespace,
+                    "MetricName": "process.memory.physical_usage",
+                    "Dimensions": [
+                        {
+                            "Name": "process.command_line",
+                            "Value": args.process_command_line_dimension_value,
+                        }
+                    ],
+                },
+                "Stat": METRIC_DATA_STATISTIC,
+                "Period": args.metrics_period,
+            },
+            "Label": "Physical Memory",
+            "ReturnData": False,
+        },
+        {
+            "Id": "total_memory_expr",
+            "Expression": "SUM([virtual_memory_raw])",
+            "Label": "Total Memory",
+            "ReturnData": True,
+            "Period": args.metrics_period,
+        },
+    ]
+
+    aws_client = boto3.client("cloudwatch")
+
+    # Create Alarms
+
+    aws_client.put_metric_alarm(
+        **{
+            **COMMON_ALARM_API_PARAMETERS,
+            "AlarmName": CPU_LOAD_ALARM_NAME,
+            "AlarmDescription": "Triggers when the CPU Load Percentage spikes above the allowed threshold DURING the Performance Test.",
+            "Threshold": args.cpu_load_threshold,
+            "Metrics": cpu_load_metric_data_queries,
```

Comment:
if we share one AWS account to 2 language SDKs for running the soaking test. Will it cause the conflict with the same metric name? ADOT usually put a random num on metric name to avoid the conflict.

Reply:
It should be okay because we are send the metrics to different namespaces so [the docs say they are isolated from eachother](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Namespace):

```yaml
LOGS_NAMESPACE: ${{ github.repository }}/soak-tests-${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}
```

If we ran 2 Soak Tests from the SAME repo that would be a problem because they _would_ conflict. But at that point we just need to add `-$GITHUB_RUN_ID` to the `LOGS_NAMESPACE`. But I don't think we need to run Soak Tests manually let alone multiple at the same time.